src/Time/Rational.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE ExplicitNamespaces   #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Time.Rational
+       ( Rat
+       , type (%)
+       , DivRat
+       , Gcd
+       , Normalize
+       ) where
+
+import GHC.TypeNats
+
+-- | Data structure represents the rational number.
+data Rat = MkRat Nat Nat
+
+type (%) = 'MkRat
```

Comment:
I thought about it a bit more.

A better name for `MkRat` is `:%`, and `%` should also perform normalization:

```
data Rat = Nat :% Nat

type family (a :: Nat) % (b :: Nat) :: Rat where
  a % b = Normalize (a :% b)
```